### PR TITLE
Add Draft of Aim 1

### PR DESCRIPTION
**Pull request by gwaybio**
Adding rough draft of aim 1. Open for comments/suggestions. We'll almost certainly have to cut some of this out.

### CODE REVIEW

File: aims/aim1.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Aim 1:
+
+**Develop proof-of-concept deep learning methods to learn a latent space from single cell transcriptomic data.**
+
+## Background
+
+Deep unsupervised models can learn to approximate a data generating manifold often represented in a lower dimensional latent space.
+The latent space of such models, including Restricted Boltzmann Machines (RBMs), Variational Autoencoders (VAEs), and Generative Adversarial Networks (GANs), preserves non-linear distances between objects and learns to represent features driving variation in the input data [1,2,3].
```
Comment (cgreene) [inline aims/aim1.md]: `features that drive`?

File: aims/aim1.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Thus, the algorithms serve a dual purpose: (1) non-linear dimensionality reduction and (2) the ability to generate hypothetical data.
```
Comment (cgreene) [inline aims/aim1.md]: I think that `data generating manifold` may be too much jargon (first sentence). Someone who misses that is likely to be confused about 2 here. Do you just want to assert these two things in the first sentence and skip the `data generating manifold` terminology there?

File: aims/aim1.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Applied to image data, the algorithms have shown much promise in generating realistic looking fake data and learning a lower dimensional representation that captures the underlying structure of high dimensional arrangements of pixels.
```
Comment (cgreene) [inline aims/aim1.md]: Example?

File: aims/aim1.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Similarly, as applied to single cell gene expression data, such algorithms can learn complex biological features that enable cell-type segmentation and cell state interpolation.
+Previously, we have demonstrated the ability to train a VAE on bulk gene expression data to identify latent features with biological meaning [4].
```
Comment (cgreene) [inline aims/aim1.md]: maybe "latent biological features"?

File: aims/aim1.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+**Develop proof-of-concept deep learning methods to learn a latent space from single cell transcriptomic data.**
+
+## Background
+
+Deep unsupervised models can learn to approximate a data generating manifold often represented in a lower dimensional latent space.
+The latent space of such models, including Restricted Boltzmann Machines (RBMs), Variational Autoencoders (VAEs), and Generative Adversarial Networks (GANs), preserves non-linear distances between objects and learns to represent features driving variation in the input data [1,2,3].
+Thus, the algorithms serve a dual purpose: (1) non-linear dimensionality reduction and (2) the ability to generate hypothetical data.
+Applied to image data, the algorithms have shown much promise in generating realistic looking fake data and learning a lower dimensional representation that captures the underlying structure of high dimensional arrangements of pixels.
+Similarly, as applied to single cell gene expression data, such algorithms can learn complex biological features that enable cell-type segmentation and cell state interpolation.
```
Comment (cgreene) [inline aims/aim1.md]: Do we know this, or hope for this?

File: aims/aim1.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+Previously, we have demonstrated the ability to train a VAE on bulk gene expression data to identify latent features with biological meaning [4].
+Others have trained GANs on images of yeast cells and interpolated the learned latent space to model protein localization [5].
```
Comment (cgreene) [inline aims/aim1.md]: Maybe with the imaging portions?

File: aims/aim1.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+Therefore, the _objective of this aim_ is to train a generative model on single cell gene expression features from the HCA demonstrating a proof of concept modeling approach.
```
Comment (cgreene) [inline aims/aim1.md]: deep nn generative model? seems like you're going for that level of specificity.

File: aims/aim1.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+## Approach
+
+We will train a VAE on single cell transcriptome data as provided by the HCA.
```
Comment (cgreene) [inline aims/aim1.md]: `as provided by` -> `from`

File: aims/aim1.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+An example of such data would be provided by the laboratory of Arjun Raj.
```
Comment (cgreene) [inline aims/aim1.md]: Data generated by Arjun Raj's lab provide an example: <some description of the data here - but I don't really understand the description in the sentence below - are these single cell data?>

File: aims/aim1.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+These data are benchmark RNAseq of homogenized cell-types under various exogenous perturbations.
+Much like the augmented training of image data under arbitrary rotations and transformations, training a VAE on cell-types under multiple perturbations will learn invariant features defining cell-type.
```
Comment (cgreene) [inline aims/aim1.md]: Confused here - isn't the rotation bit the bootstrap? Might be better to have its own paragraph highlighting this as a methodological advance that can let us succeed.

File: aims/aim1.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+Aiding in our search for invariant features, we will use a fast bootstrap resampling approach developed in the laboratory of Rob Patro.
+Briefly, the approach will resample RNAseq reads from the same sample under different initialized parameters artificially increasing sample size while retaining cell-type features.
+
+In order to select the best model to validate, we will perform an extensive grid search with a held-out validation set to identify the optimal VAE architecture and hyperparameters.
+We will select optimal models based on minimizing reconstruction loss plus a KL divergence term which constrains the features to a Gaussian distribution.
+Once an optimal model is identified, we will demonstrate proof of concept by validating learned features.
+The largest features in single cell gene expression are the abundance of technical and biological zeroes, and we will assess the ability of the VAE to automatically extract this artifact by comparing it to existing batch correction methods [6].
```
Comment (cgreene) [inline aims/aim1.md]: does bootstrap resampling also help you here?

File: aims/aim1.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+Additionally, we will use the learned VAE features to classify specific perturbations.
```
Comment (cgreene) [inline aims/aim1.md]: Is this more like your aim 3? Can't tell if there's bleeding between the aims here or if this is different.

**Review by cgreene — COMMENTED** [review]
some thoughts that came to mind on a first pass

File: aims/aim1.md (hunk 2 above)
Comment (jaclyn-taroni) [reply aims/aim1.md]: I agree with this comment from @cgreene

File: aims/aim1.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+# Aim 1:
+
+**Develop proof-of-concept deep learning methods to learn a latent space from single cell transcriptomic data.**
+
+## Background
+
+Deep unsupervised models can learn to approximate a data generating manifold often represented in a lower dimensional latent space.
```
Comment (jaclyn-taroni) [inline aims/aim1.md]: Do you think you might port over some text from Aim 3 that might fit in well here? Specifically I'm thinking about the opening of your background in [Aim 3](https://github.com/greenelab/czi-rfa/blob/master/aims/aim3.md)

> Latent features describe a lower dimensional representation of input data and there are many methods, including principal components analysis (PCA) and non-negative matrix factorization (NMF), that arrive at unique solutions. Certain classes of deep generative models, including variational autoencoders (VAEs) and generative adversarial networks (GANs), that add a distribution matching constraint, imbue intuitive mathematical features to the learned latent features [1,2].

File: aims/aim1.md (hunk 6 above)
Comment (jaclyn-taroni) [reply aims/aim1.md]: Yea, I might move this to line 11 after the first mention of image data

File: aims/aim1.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+Similarly, as applied to single cell gene expression data, such algorithms can learn complex biological features that enable cell-type segmentation and cell state interpolation.
+Previously, we have demonstrated the ability to train a VAE on bulk gene expression data to identify latent features with biological meaning [4].
+Others have trained GANs on images of yeast cells and interpolated the learned latent space to model protein localization [5].
+Therefore, the _objective of this aim_ is to train a generative model on single cell gene expression features from the HCA demonstrating a proof of concept modeling approach.
+
+## Approach
+
+We will train a VAE on single cell transcriptome data as provided by the HCA.
+An example of such data would be provided by the laboratory of Arjun Raj.
+These data are benchmark RNAseq of homogenized cell-types under various exogenous perturbations.
```
Comment (jaclyn-taroni) [inline aims/aim1.md]: Would explicitly state that these are bulk. I'm also struggling a bit with how these data fit in the way this is currently written.

File: aims/aim1.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+Much like the augmented training of image data under arbitrary rotations and transformations, training a VAE on cell-types under multiple perturbations will learn invariant features defining cell-type.
+Aiding in our search for invariant features, we will use a fast bootstrap resampling approach developed in the laboratory of Rob Patro.
+Briefly, the approach will resample RNAseq reads from the same sample under different initialized parameters artificially increasing sample size while retaining cell-type features.
+
```
Comment (jaclyn-taroni) [inline aims/aim1.md]: I'm going to put some (sometimes not fully-formed) thoughts about how I would reorganize the first paragraph into its own paragraph on the bootstrap bit here:

```markdown
**sentence about data-intensive requirements of VAE and augmented training in image application**
We will use a fast bootstrap resampling approach developed in the laboratory of Rob Patro to augment training data.
Briefly, the approach will resample RNAseq reads from the same sample under different initialized parameters artificially increasing sample size while retaining cell-type features.
Much like the augmented training of image data under arbitrary rotations and transformations, training a VAE on bootstrapped cells will learn invariant features defining cell type.
```

File: aims/aim1.md (hunk 11 above)
Comment (jaclyn-taroni) [reply aims/aim1.md]: Are you thinking you might do a (bootstrap resampling-aided?) simulation experiment here?

File: aims/aim1.md (hunk 12 above)
Comment (jaclyn-taroni) [reply aims/aim1.md]: I agree that this might be getting close to an alternative approach for aim 3

**Review by jaclyn-taroni — COMMENTED** [review]
Some thoughts about organization @gwaygenomics

File: aims/aim1.md (hunk 2 above)
Comment (huqiwen0313) [inline aims/aim1.md]: What does it mean by "the ability to generate hypothetical data" ?

File: aims/aim1.md (hunk 5 above)
Comment (gwaybio) [reply aims/aim1.md]: I think "hypothesize" is a better way to put it

File: aims/aim1.md (hunk 2 above)
Comment (gwaybio) [reply aims/aim1.md]: Yeah, too much jargon as is. I've kept this sentence in tact, but reworded the first to introduce data generation more explicitly.

File: aims/aim1.md (hunk 6 above)
Comment (gwaybio) [reply aims/aim1.md]: good thinking - works better there

File: aims/aim1.md (hunk 12 above)
Comment (gwaybio) [reply aims/aim1.md]: based on discussions this afternoon, I am going to cut this eval

File: aims/aim1.md (hunk 13 above)
Comment (gwaybio) [reply aims/aim1.md]: I am not sure. Based on how the aim 1 is structured in this PR I think it makes sense - but in an updated version of the aim, I don't think it does. My current thinking is:

aim 1 - train/troubleshoot the model
aim 3 - latent space arithmetic (which necessitates an eval to other latent space finding algorithms)

File: aims/aim1.md (hunk 2 above)
Comment (gwaybio) [reply aims/aim1.md]: yeah, this is confusing. I've updated the structure of the first three sentences to have better context to introducing point (2) here - which I've also updated to `(2) the ability to interpolate between data.`. Changes in the next commit!

File: aims/aim1.md (hunk 9 above)
Comment (gwaybio) [reply aims/aim1.md]: No - they're homogenized cell-types in bulk. Perhaps I could move a description of this data to the eval section. It feels kinda forced here

File: aims/aim1.md (hunk 14 above)
Comment (gwaybio) [reply aims/aim1.md]: yeah, right now it feels forced. Going to move to an eval - I think it fits nicely there

File: aims/aim1.md (hunk 11 above)
Comment (gwaybio) [reply aims/aim1.md]: I think this may fit better under potential benefits of the approach rather than as an eval

**Comment by gwaybio** [comment]
thanks for comments @cgreene @jaclyn-taroni @huqiwen0313 - I think its ready for another look

File: aims/aim1.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+In order to select the best model to validate, we will perform an extensive grid search with a held-out validation set to identify optimal VAE architecture and hyperparameters.
+We will select optimal models based on minimizing reconstruction loss plus a KL divergence term which constrains the features to a Gaussian distribution.
+
+In addition, we will compare our optimized VAE to an alternative VAE we will train with an augmentation approach.
+A common image processing goal is to disentangle invariant features from a corpus of images.
+These invariant features are key, but the algorithms must learn how to extract them under various rotations, zoom, and other augmentations.
+Images can be resampled with artificial rotation and added back as input to deep models to encourage invariant feature learning.
+We will collaborate with Rob Patro's laboratory and pursue a similar approach applied to single cell gene expression: Rapid bootstrapping and subsampling of raw RNAseq reads to overcome transcript depth and rare transcript biases.
+
+We will evaluate our models using different transcriptome data from the HCA.
```
Comment (jaclyn-taroni) [inline aims/aim1.md]: `transcriptomic` vs. `transcriptome` -- not sure what is "more correct"

File: aims/aim1.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+Therefore, the _objective of this aim_ is to train a deep neural network generative model built from single cell gene expression from the HCA demonstrating a proof of concept modeling approach.
+
+## Approach
+
+We will train a VAE on single cell transcriptome data from the HCA.
+In order to select the best model to validate, we will perform an extensive grid search with a held-out validation set to identify optimal VAE architecture and hyperparameters.
+We will select optimal models based on minimizing reconstruction loss plus a KL divergence term which constrains the features to a Gaussian distribution.
+
+In addition, we will compare our optimized VAE to an alternative VAE we will train with an augmentation approach.
+A common image processing goal is to disentangle invariant features from a corpus of images.
```
Comment (jaclyn-taroni) [inline aims/aim1.md:23]: I still think you might need a sentence that provides the intuition behind invariant features. I'm also looking for the _why_ here.

File: aims/aim1.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+These invariant features are key, but the algorithms must learn how to extract them under various rotations, zoom, and other augmentations.
+Images can be resampled with artificial rotation and added back as input to deep models to encourage invariant feature learning.
```
Comment (jaclyn-taroni) [inline aims/aim1.md]: Citation?

File: aims/aim1.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
+# Aim 1:
+
+**Develop proof-of-concept deep learning methods to learn a latent space from single cell transcriptomic data.**
+
+## Background
```
Comment (jaclyn-taroni) [inline aims/aim1.md:5]: Nice changes to this section 👍

**Review by jaclyn-taroni — COMMENTED** [review]
Couple minor comments about **Approach** @gwaygenomics

File: aims/aim1.md (hunk 16 above)
Comment (gwaybio) [reply aims/aim1.md]: probably transcriptomic here, must have overlooked it

File: aims/aim1.md (hunk 17 above)
Comment (gwaybio) [reply aims/aim1.md:23]: agree - I try to clarify in the next commit

Comment (gwaybio) [reply aims/aim1.md:23]: I modified some of the subsequent sentences, hopefully clarifying some of the intuition

**Comment by cgreene** [comment]
I'm going to propose some revisions as I integrate these today, but will do that through a separate PR for the aims. Thanks!

**Pull request merged**